tap/extensions/amqp/read.go

```diff
@@ -299,7 +299,7 @@ func readArray(r io.Reader) ([]interface{}, error) {
 
 	var (
 		lim   = &io.LimitedReader{R: r, N: int64(size)}
-		arr   = []interface{}{}
+		arr   []interface{}
```

Comment:
not sure this is the best way, and it could break something (there is a difference between empty array and nil)

Reply:
append is working when nil, so it is ok to create it as nil..